### ISSUE TITLE
Unnecessarily long function

### ISSUE THREAD

**Issue by fayssalchokr** [ennisme/ayaos#14]
The function kernel::runtime::ensure_loaded_correctly() has an unnecessarily long if statement that can be replaced with a more suitable function like kernel::compare_memory().

**Comment by ennisme** [comment]
You're right. 🙂 Reason is ensure_loaded_correctly was introduced very early on before compare_memory was even a thing.
